yunwoo/baekjoon/스택/1874-스택-수열.js

```diff
@@ -0,0 +1,48 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const n = +input[0];
+  const stack = [];
+  const targetSequence = input.slice(1);
+  let count = 1;
+  let result = [];
+
+  for (let i = 0; i < n; i++) {
+    const target = targetSequence[i];
+
+    while (count <= +target) {
+      stack.push(`${count}`);
+      result.push("+");
+      count++;
+    }
+
+    const lastElement = stack.pop();
+    result.push(`-`);
+
+    if (lastElement !== target) {
+      result = ["NO"];
+      break;
+    }
+  }
+
+  console.log(result.join("\n"));
```

Comment:
```suggestion
  const n = +input[0];
  const stack = [];
  const targetSequence = input.slice(1).map(Number);
  let count = 1;
  let result = [];

  for (let i = 0; i < n; i++) {
    const target = targetSequence[i];

    while (count <= target) {
      stack.push(count);
      result.push("+");
      count++;
    }

    const lastElement = stack.pop();
    result.push(`-`);

    if (lastElement !== target) {
      result = ["NO"];
      break;
    }
  }

  console.log(result.join("\n"));
```
- 풀이 잘봤습니다 ~
- 미리 입력값을 형변환 해주세요 ! 
- 어떤 거는 +target 숫자로 형변환하고 \`${count}\` 로 문자열로 형변환 하고 조금 뒤죽박죽입니다 ~ 애초에 처음에 형변환을 해놓고 활용했으면 이런 부분들을 개선할 수 있씁니다.

Reply:
- 처음엔 map을 돌리는게 효율에서 떨어질거라 생각하고 안했었는데 보고나니 확실히 가독성이나 코드를 읽을 때 별로네요ㅎㅎ..
- 코멘트 감사합니다!!!🙇‍♂️